test/libsolidity/syntaxTests/parsing/constructor_super.sol

```diff
@@ -0,0 +1,19 @@
+contract A {
+  function x() pure internal {}
+}
+
+contract B is A {
+  constructor() public {
+    super.x();
+  }
+
+  function x() pure internal {
+    require(false);
```

Comment:
Please reduce the test case to its bare minimum.

Reply:
And perhaps also add a comment that this tests a former bug in the detection of `this.f()`.